tests/test_routing.py

```diff
@@ -1164,6 +1162,65 @@ def test_extract_straher_streams_watersheds_d8(self):
         watershed_vector = None
         watershed_layer = None
 
+    def test_single_drain_point(self):
+        """PGP.routing: test single_drain_point pitfill."""
+        dem_array = numpy.zeros((11, 11), dtype=numpy.float32)
+        dem_array[0, 0] = -1.0
+        dem_array[1:8, 1:8] = -2.0
+        dem_array[10, 7] = -4
+        dem_array[8:11, 8:11] = 2.0
+        dem_array[9:11, 9:11] = 1.0
+        dem_array[10, 10] = -7.0
+        dem_path = os.path.join(self.workspace_dir, 'dem.tif')
+        _array_to_raster(dem_array, None, dem_path)
+
+        # outlet tuple at 0,0, just drain one edge
+        expected_array_0_0 = numpy.copy(dem_array)
+        expected_array_0_0[1:8, 1:8] = -1
+        expected_array_0_0[10, 7] = 0
+        expected_array_0_0[8:11, 8:11] = 2.0
+
+        # output tuple at 5,5, it's a massive pit so drain the edges
+        expected_array_5_5 = numpy.copy(dem_array)
+        expected_array_5_5[8:11, 8:11] = 2.0
+        expected_array_5_5[10, 7] = 0
```

Comment:
We're just testing the algorithm works by draining everything to Sarlacc Pit?

Reply:
GOOD ONE